Updated README - see description (#49)
UBC-MDS/samplingsimulatorpy: README.md

`samplingsimulatorpy` is a Python package intended to assist those teaching or learning basic statistical inference.

### Authors

| Name             | GitHub                                          |
| ---------------- | ----------------------------------------------- |
| Holly Williams   | [hwilliams10](https://github.com/hwilliams10)   |
| Lise Braaten     | [lisebraaten](https://github.com/lisebraaten)   |
| Tao Guo          | [tguo9](https://github.com/tguo9)               |
| Yue (Alex) Jiang | [YueJiangMDSV](https://github.com/YueJiangMDSV) |

### Overview

This package allows users to generate virtual populations which can be sampled from in order to compare and contrast sample vs sampling distributions for different sample sizes.  The package also allows users to sample from the generated virtual population (or any other population), plot the distributions, and view summaries for the parameters of interest.

## Installation:

```
pip install -i https://test.pypi.org/simple/ samplingsimulatorpy
```

## Function Descriptions

- `generate_virtual_pop` creates a virtual population.

>     - **Inputs** : distribution function (i.e. `np.random.lognormal`, `np.random.binomial`, etc), the paramaters required by the distribution function, and the size of the population.

Reviewer: ```suggestion
    - **Inputs** : distribution function (i.e. `np.random.lognormal`, `np.random.binomial`, etc), the parameters required by the distribution function, and the size of the population.
```

 Python `pandas` already includes some summary statistics functions such as `.describe()`, however our package will be more customizable.  Our summary will only include the statistical parameters of interest and will provide a comparison between the sample, sampling, and true population parameters.

### Dependencies

- python = "^3.7"
- pandas = "^1.0.1"
- numpy = "^1.18.1"
- altair = "^4.0.1"
- flake8 = "^3.7.9"

## Usage

#### `generate_virtual_pop`

``` 
from samplingsimulatorpy import generate_virtual_pop
generate_virtual_pop(size, distribution_func, *para)
```

**Arguments:**

  - `size`: The number of samples
  - `distribution_func`: The distribution that we are generating samples from
  - `*para`: The arguments required for the distribution function

**Example:**

`pop <- generate_virtual_pop(100, "height", rnorm, 0, 1)`

> #### `draw_samples`

Reviewer: seems to be R syntax
```suggestion
`pop = generate_virtual_pop(100, np.random.normal, 0, 1)`

#### `draw_samples`
```

``` 
from samplingsimulatorpy import draw_samples
draw_samples(pop, reps, n_s)
```

**Arguments:**

  - `pop` the virtual population as a tibble
  - `reps` the number of replication for each sample size as an integer
    value

>   - `n_s` the sample size for each one of the samples as an array

Reviewer: ```suggestion
  - `pop` the virtual population as a data frame
  - `reps` the number of replication for each sample size as an integer
    value
  - `n_s` the sample size for each one of the samples as a list
```

**Example:**

> `samples <- draw_samples(pop, 3, [5, 10, 15, 20])`

Reviewer: ```suggestion
`samples = draw_samples(pop, 3, [5, 10, 15, 20])`
```

#### `plot_sample_hist`

``` 
from samplingsimulatorpy import plot_sample_hist
plot_sample_hist(pop, samples)
```

**Arguments:**

  - `pop` the virtual population as a tibble

>   - `samples` the samples as a tibble

Reviewer: ```suggestion
  - `pop` the virtual population as a data frame
  - `samples` the samples as a data frame
```

**Example:**

`plot_sample_hist(samples)`

#### `plot_sampling_hist`

``` 
from samplingsimulatorpy import plot_sampling_hist
plot_sampling_hist(pop, samples)
```

**Arguments:**

>   - `samples` the samples as a tibble

Reviewer: ```suggestion
  - `samples` the samples as a data frame
```
